### PR TITLE
feat: Search improvements for conversations 

### PR DESCRIPTION
**Pull request by nithindavid**
## Description

Fixes part of issues mentioned in https://github.com/chatwoot/chatwoot/issues/6583

**Note** Moving the search sidebar needs more brainstorming as we have expand and sidebar toggle icons. Will take this as a separate ticket.

cc: @pranavrajs 

image

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/javascript/dashboard/modules/search/components/SearchResultConversationsList.vue [hunk 1]
```diff
@@ -4,7 +4,7 @@
     :empty="!conversations.length"
     :query="query"
   >
-    <ul class="search-list">
+    <ul class="search-list" @click="closeSearch">
```
Comment (scmmishra) [inline app/javascript/dashboard/modules/search/components/SearchResultConversationsList.vue:7]: Should this be on the individual `<li>` elements instead?

File: app/javascript/dashboard/modules/search/components/SearchView.vue [hunk 2]
```diff
@@ -1,46 +1,62 @@
...
+            <search-result-messages-list
+              v-if="filterMessages"
+              :messages="messages"
+              :query="query"
+              @close-search="closeSearch"
+            />
+            <search-result-conversations-list
+              v-if="filterConversations"
+              :conversations="conversations"
+              :query="query"
```
Comment (scmmishra) [inline app/javascript/dashboard/modules/search/components/SearchView.vue:39]:
```suggestion
              :query="query"
              @close-search="closeSearch"
```

Seems like you missed the listener here

File: app/javascript/dashboard/modules/search/components/SearchView.vue [hunk 3]
```diff
@@ -1,46 +1,62 @@
...
+            </p>
+          </div>
+          <div v-else class="empty text-center">
+            <p class="text-center">
+              <fluent-icon icon="search" size="24px" class="icon" />
+            </p>
+            <p class="empty-state__text">
+              {{ $t('SEARCH.EMPTY_STATE_DEFAULT') }}
+            </p>
+          </div>
```
Comment (scmmishra) [inline app/javascript/dashboard/modules/search/components/SearchView.vue:55]: This does not trigger correctly when clearing the input.

Steps
1. Search a string that will result an empty search
2. Delete the input

The empty state is the one of "no result found" for the previous search

image

File: app/javascript/dashboard/modules/search/components/SearchView.vue [hunk 4]
```diff
@@ -147,34 +166,54 @@ export default {
...
-.search-root {
-  max-width: 800px;
+.backdrop {
+  position: fixed;
+  top: 0;
   width: 100%;
-  margin: 0 auto;
   height: 100%;
+  background: var(--white-transparent);
+  z-index: var(--z-index-very-high);
```
Comment (scmmishra) [inline app/javascript/dashboard/modules/search/components/SearchView.vue:196]: The backdrop does not apply to the entire screen, only to the main conversation area. Is this intentional?

image

**Review by scmmishra — COMMENTED** [review]
Added some suggestions, a few other points.

1. the `/` key binding does not work when the search box is closed. Should we enable it anyway on the conversation screen
2. Not sure if it is in the scope of this PR, we can maybe take a look at it separately, but keyboard navigation in general is broken for the layout. 
    1. The search tabs are not focusable using the `Tab Button`
    2. The focus state is not handled for search items. image
    3. There's no way to dismiss the search dialog, say using `Esc` key

Some UI nitpicks. These are only changes from an aesthetics point of view. Does not improve or affect usability of functionality. 
Leaving this upto your discretion, feel free to address them as you see fit. 

#### The `x` axis padding seems to be quite inconsistent across the board.

Group 1

We can remove the margin of the `tab-container` and make the bottom border extend end to end. And instead of aligning the search list item padding to the container, we can align it's content to it

#### We have a dedicated null state for search results that yield no results. 

image 

I think it would be better to use the empty state of the individual tabs, instead of this. That way the state is more representative of the actual UI and how the data will be presented

**Review by scmmishra — APPROVED** [review]
LGTM!

**Comment by pranavrajs** [comment]
Screenshot 2023-03-14 at 10 01 07 AM


@scmmishra @nithindavid Can we please re-consider this change. I don't see why we cannot utilize the white space around page. 

Screenshot 2023-03-14 at 10 01 58 AM

The previous one was better.

**Comment by nithindavid** [comment]
@pranavrajs I'll iterate on this feedback 👍

**Pull request merged**